testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/problems/KnownProblemIds.groovy

```diff
@@ -25,13 +25,15 @@ class KnownProblemIds {
             def pattern = it.key
             definition.id.fqid ==~ pattern
```

Comment:
You can also just check for a pattern, instead of for an exact value, if needed.

Reply:
You mean in line 30, the definition could be:
```java
        def knownDisplayName = knownDefinition.find { it ==~ definition.id.displayName }

```

Reply:
Because in that case, I think I would wait until the use case arrives.
I'm going to merge this in, but if you feel strongly about this, I'm okay doing this change later.